packages/shared/src/effect.ts

```diff
@@ -50,6 +50,40 @@ export const fetchEff = (
     }),
   );
 
+// TODO: rename the other one to fetchEffJsonSchema and this to fetchEffJson
+// though generally I would avoid XandY kind utils in favor of composition
+export const fetchEffUnknown = (
+  input: RequestInfo | URL,
+  /** Schema to be used if the response returned a 2xx  */
+  init?: RequestInit,
+): Effect.Effect<unknown, FetchError, FetchContext> => {
+  const requestUrl = input instanceof Request ? input.url : input.toString();
+
+  return fetchEff(input, init).pipe(
+    Effect.andThen((res) =>
+      Effect.tryPromise({
+        try: async () => {
+          const json = await res.json();
+          return { ok: res.ok, json, status: res.status };
+        },
+        catch: (error) => new FetchError({ error, input }),
+      }),
+    ),
+    Effect.filterOrFail(
+      (_) => _.ok,
+      ({ json, status }) =>
+        new FetchError({
+          error: `Request to ${requestUrl} failed with status ${status}`,
+          data: json as Json,
+          input,
+        }),
+    ),
+    Effect.withSpan("fetchRawJson", {
+      attributes: { input: JSON.stringify(input) },
+    }),
+  );
+};
```

Comment:
it would be best if the util could be broken into multiple components, when faced with `fetchAndParseJsonAndDecode` it's hard to later pick-and-choose which parts I want.
While it's a bit more annoying to have to do
```ts
fetch(...).pipe(
  Effect.flatMap(json),
  Effect.flatMap(decode)
)
```
it beings dividends as each piece is more useful on its own.

In this case, I wanted only the `fetch` part and the logic wrapped inside the `andParseJson` part, but without the `andDecode` part 😢

Reply:
i think that makes sense 👍🏼